presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1626,25 +1654,24 @@ private void analyzeAggregations(QuerySpecification node,
             }
         }
 
-        ImmutableList<FieldOrExpression> allGroupingColumns = groupingSets.stream()
-                .flatMap(Collection::stream)
-                .distinct()
-                .collect(toImmutableList());
-
         // is this an aggregation query?
-        if (!aggregates.isEmpty() || !allGroupingColumns.isEmpty()) {
+        if (!aggregates.isEmpty() || !(groupingSets.size() == 1 && getOnlyElement(groupingSets).isEmpty())) {
```

Comment:
Sorry, it's not clear to me why the existing line was incorrect:
- allGroupingColumns in the previous PR is distinctGroupingColumns in this one
- so the check was (a) if there was an implicit aggregation, (b) there was an explicit non-empty aggregation
- ``GROUP BY ()`` doesn't make any sense without an implicit aggregation, so the if would do exactly what I would expect.